### PR TITLE
Add HTML form elements browser compat data (except for <input>)

### PR DESCRIPTION
**Pull request by teoli2003**
Compat data for: &lt;datalist&gt;, &lt;data&gt;, &lt;fieldset&gt;, &lt;form&gt;, &lt;keygen&gt;, &lt;label&gt;, &lt;legend&gt;, &lt;meter&gt;, &lt;optgroup&gt;, &lt;option&gt;, &lt;output&gt;, &lt;progress&gt;, &lt;select&gt;, &lt;textarea&gt;.

&lt;input&gt; will have its own PR>

### CODE REVIEW

**Comment by Elchi3** [comment]
<data> is already done, hence the conflict.

**Comment by teoli2003** [comment]
Removed and rebased.

File: html/elements/select.json [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+            },
+            "firefox": {
+              "version_added": "1",
+              "notes": "Historically, Firefox has allowed keyboard and mouse events to bubble up from the <code>&lt;option&gt;</code> element to the parent <code>&lt;select&gt;</code> element, although this behavior is inconsistent across many browsers. For better Web compatibility (and for technical reasons), when Firefox is in multi-process mode the <code>&lt;select&gt;</code> element is displayed as a drop-down list. The behavior is unchanged if the <code>&lt;select&gt;</code> is presented inline and it has either the multiple attribute defined or a size attribute set to more than 1. Rather than watching <code>&lt;option&gt;</code> elements for events, you should watch for change events on <code>&lt;select&gt;</code>. See <a href='https://bugzil.la/1090602'>bug 1090602</a> for details."
+            },
+            "firefox_android": {
+              "version_added": "1",
+              "notes": "Firefox for Android, by default, sets a <code>background-image</code> gradient on all <code>&lt;select multiple&gt;</code> elements. This can be disabled using <code>background-image: none<code>."
```
Comment (Elchi3) [inline html/elements/select.json]: The last `<code>` should be `</code>`.

File: html/elements/textarea.json [hunk 2]
```diff
@@ -0,0 +1,869 @@
...
+            "edge": {
+              "version_added": true
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": true,
+              "notes": [
+                "Before Firefox 6.0 , when a <code>&lt;textarea&gt;</code> was focused, the insertion point was placed at the end of the text by default. Other major browsers place the insertion point at the beginning of the text.",
```
Comment (Elchi3) [inline html/elements/textarea.json]: Nit: "Before Firefox 6," (no space between 6 and comma, no leading zero)

File: html/elements/textarea.json [hunk 3]
```diff
@@ -0,0 +1,869 @@
...
+                },
+                "edge": {
+                  "version_added": true
+                },
+                "edge_mobile": {
+                  "version_added": true
+                },
+                "firefox": {
+                  "version_added": false,
+                  "notes": "See <a href='https://bugzil.la/1312897'>bug 1312897</a>"
```
Comment (Elchi3) [inline html/elements/textarea.json]: Notes should always end with a full stop.

File: html/elements/form.json [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+      "form": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/HTML/Element/form",
+          "support": {
+            "webview_android": {
+              "version_added": true
+            },
+            "chrome": {
+              "version_added": true,
+              "notes": "The Google Chrome UI for auto-complete request varies, depending on whether <code>autocomplete</code> is set to <code>off</code> on <code>&lt;input&gt;</code> elements as well as their form. Specifically, when a form has <code>autocomplete</code> set to <code>off</code> and its <code>&lt;input&gt;</code> element's <code>autocomplete</code> fttribute is not set, then if the user asks for autofill suggestions for the <code>&lt;input&gt;</code> element, Chrome might display a message saying 'autocomplete has been disabled for this form.' On the other hand, if both the form and the input element have <code>autocomplete</code> set to <code>off</code>, the browser will not display that message. For this reason, you should set <code>autocomplete</code> to <code>off</code> for each <code>&lt;input&gt;</code> that has custom auto-completion."
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/form.json]: Fixed.

File: html/elements/optgroup.json [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                "version_added": true
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          },
+          "label": {
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/optgroup.json]: Fixed.

File: html/elements/option.json [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+              "edge": {
+                "version_added": true
+              },
+              "edge_mobile": {
+                "version_added": true
+              },
+              "firefox": {
+                "version_added": "1",
+                "notes": [
+                  "Gecko doesn't display the value of the <code>label</code> attribute as option text if element's content is empty. See <a href='https://bugzil.la/40545'>bug 40545</a>.",
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/option.json]: Fixed.

File: html/elements/option.json [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+                  "Historically, Firefox has allowed keyboard and mouse events to bubble up from the <code>&lt;option&gbt;</code> element to the parent <code>&lt;select&gt;</code> element, although this behavior is inconsistent across many browsers. For better Web compatibility (and for technical reasons), when Firefox is in multi-process mode and the <code>&lt;select&gt;</code> element is displayed as a drop-down list. The behavior is unchanged if the <code>&lt;select&gt;</code> is presented inline and it has either the <code>multiple</code> attribute defined or a <code>size</code> attribute set to more than <code>1</code>. Rather than watching <code>option</code> elements for events, you should watch for <a href='https://developer.mozilla.org/en-US/docs/Web/Events/change'>change</a> events on <code>&lt;select&gt;</code>. See <a href='https://bugzil.la/1090602'>bug 1090602</a> for details.",
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/option.json]: Fixed (both)

File: html/elements/progress.json [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+              "notes": [
+                "Prior to Firefox 14, the <code>&lt;progress&gt;</code> element was incorrectly classified as a form element, and therefore had a <code>form</code> attribute. This has been fixed.",
+                "Gecko provides the <code>::-moz-progress-bar</code> pseudo-element, which lets you style the part of the interior of the progress bar representing the amount of work completed so far."
+              ]
+            },
+            "firefox_android": {
+              "version_added": "6",
+              "notes": [
+                "Prior to Firefox 14, the <code>&lt;progress&gt;</code> element was incorrectly classified as a form element, and therefore had a <code>form</code> attribute. This has been fixed.",
+                "Gecko provides the <code>::-moz-progress-bar</code> pseudo-element, which lets you style the part of the interior of the progress bar representing the amount of work completed so far."
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/progress.json]: Fixed.

File: html/elements/select.json [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+    "elements": {
+      "select": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/HTML/Element/select",
+          "support": {
+            "webview_android": {
+              "version_added": true,
+              "notes": [
+                "In the Browser app for Android 4.1 (and possibly later versions), there is a bug where the menu indicator triangle on the side of a <code>&lt;select&gt;</code> will not be displayed if a <code>background</code>, <code>border</code>, or <code>border-radius</code> style is applied to the <code>&lt;select&gt;</code>.",
+                "Ignore <code>border-radius</code> on <code>&lt;select&gt;</code> elements unless <code>-webkit-appearance</code> is overridden to an appropriate value."
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/select.json]: Fixed.

File: html/elements/select.json [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+              ]
+            },
+            "chrome": {
+              "version_added": true,
+              "notes": "Ignore <code>border-radius</code> on <code>&lt;select&gt;</code> elements unless <code>-webkit-appearance</code> is overridden to an appropriate value."
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/select.json]: Fixed.

File: html/elements/select.json [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+            },
+            "chrome_android": {
+              "version_added": true,
+              "notes": "Ignore <code>border-radius</code> on <code>&lt;select&gt;</code> elements unless <code>-webkit-appearance</code> is overridden to an appropriate value."
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply html/elements/select.json]: Fixed (3 times)

File: html/elements/select.json (hunk 1 above)
Comment (teoli2003) [reply html/elements/select.json]: Fixed.

File: html/elements/textarea.json (hunk 2 above)
Comment (teoli2003) [reply html/elements/textarea.json]: Fixed (twice)

File: html/elements/textarea.json (hunk 3 above)
Comment (teoli2003) [reply html/elements/textarea.json]: Fixed (twice)

**Comment by teoli2003** [comment]
@Elchi3 Fixed all review comments.

**Pull request merged**